### ISSUE TITLE
Do tool checking in cWorld

### ISSUE THREAD

**Issue by NiLSPACE** [cuberite/cuberite#119]
The tool checking (e. g. diamonds can be mined by iron pickaxe or better, etc.) is done in the cClientHandle/cPlayer. It should be done in cWorld so that plugins can use it as well. Or provide a documented way of doing this using a common code path.

http://forum.mc-server.org/showthread.php?tid=434&pid=9560#pid9560

**Comment by SamJBarney** [comment]
Is this issue valid anymore?

**Comment by NiLSPACE** [comment]
I believe it still is.

**Comment by worktycho** [comment]
It should not be done in cClientHandle, but cWorld is not a good place either. I suspect we need to create a new object to handle player policy, similar to how the blockhandlers handle block policy.

**Comment by NiLSPACE** [comment]
I think it makes more sense to do it in cWorld. Then you can create a function that looks like this:
```cpp
void cWorld::DigBlockWithTool(int a_X, int a_Y, int a_Z, BLOCKTYPE a_DigItem)
{
   cItems Pickups = cItems();
   SpawnItemPickups(
      a_X, a_Y, a_Z, 
      cBlockHander::CreateBlockHandler(GetBlock(a_X, a_Y, a_Z)).ConvertToPickups(Pickups, GetBlockMeta(a_X, a_Y, a_Z)
   );
}
```

**Comment by SphinxC0re** [comment]
So should we create a ```cPlayerHandle``` class to handle all the player interactions in a more or less abstract way?

**Comment by SphinxC0re** [comment]
@NiLSPACE what do you think?

**Comment by NiLSPACE** [comment]
I guess so, but I fail to see how it would all interact with each other. Which method would call what method. Could someone clarify?

**Comment by lighth7015** [comment]
On a related note, I've crashed my development copy of Cuberite by trying to worldedit a section of blocks into diamond drops because I forgot what item id# diamond blocks were.. Oops.

We should really ensure things like that are caught before it gets to the "Oh, shit.. abort()!" stage. :p

**Comment by mathiascode** [comment]
Implemented in #4417
